File: withoutSWIG-C/example_main.lua
#!/usr/bin/luajit

function main()
    local ffi = require ('ffi')
    ffi.cdef[[
        int Example(char *s);
        char *Get(int id);
    ]]

    local c_example = ffi.load('./c_example.so')

    d = ffi.new("char[200]")
    ffi.copy(d, 'teszt-A'); a = c_example.Example(d)
    ffi.copy(d, 'teszt-B'); b = c_example.Example(d)

    print (ffi.string(c_example.Get(a)))
    print (ffi.string(c_example.Get(b)))
    print (ffi.string(c_example.Get(a)))
    print (ffi.string(c_example.Get(a)))
    print (ffi.string(c_example.Get(a)))
    print (ffi.string(c_example.Get(a)))
    print (ffi.string(c_example.Get(b)))
end

main()

File: Rust-Python-class/example_main.py
#!/usr/bin/python3

import rust_example

if __name__ == '__main__':
    a = rust_example.Example('teszt-A')
    b = rust_example.Example('teszt-B')

    print (a.get())
    print (b.get())
    print (a.get())
    print (a.get())
    print (a.get())
    print (a.get())
    print (b.get())
